### PR TITLE
docs: FAQ doc added make deps execution failed

### LINKED ISSUE #3632: docs: the FAQ document adds the issue of failure to execute the make deps command
# Improve Docs

When deploying APISIX in source code mode, if the `openresty-openssl111-devel` dependency is missing, the following error message will occur when executing `make deps`:

```shell
$ make deps
......
luasocket 3.0rc1-2 is now installed in /home/apisix/deps (license: MIT)

apisix master-0 depends on luasec 0.9-1 (not installed)
Installing https://luarocks.org/luasec-0.9-1.src.rock

Error: Failed installing dependency: https://luarocks.org/luasec-0.9-1.src.rock - Could not find header file for OPENSSL
  No file openssl/ssl.h in /usr/local/openresty/openssl111/include
You may have to install OPENSSL in your system and/or pass OPENSSL_DIR or OPENSSL_INCDIR to the luarocks command.
Example: luarocks install luasec OPENSSL_DIR=/usr/local
make: *** [deps] error 1
```

For such common problems, we should record such problems in the FAQ.

### PR DESCRIPTION
**Pull request by Firstsawyou**
### What this PR does / why we need it:

close #3632

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -355,3 +355,19 @@ The high availability of APISIX can be divided into two parts:
 1. The data plane of Apache APISIX is stateless and can be elastically scaled at will. Just add a layer of LB in front.
 
 2. The control plane of Apache APISIX relies on the highly available implementation of `etcd cluster` and does not require any relational database dependency.
+
+## Why does the `make deps` command fail in source installation?
+
+When executing the `make deps` command, an error such as the one shown below occurs. This is caused by the missing dependency of `openresty-openssl111-devel`, you need to execute `yum install -y openresty-openssl111-devel` to install it first.
```
Comment (tokers) [inline FAQ.md:361]: Well, people might not use RedHat series Linux distro.

Comment (Firstsawyou) [reply FAQ.md]: updated.

File: FAQ.md [hunk 2]
```diff
@@ -358,7 +358,7 @@ The high availability of APISIX can be divided into two parts:
 
 ## Why does the `make deps` command fail in source installation?
 
-When executing the `make deps` command, an error such as the one shown below occurs. This is caused by the missing dependency of `openresty-openssl111-devel`, you need to execute `yum install -y openresty-openssl111-devel` to install it first.
+When executing the `make deps` command, an error such as the one shown below occurs. This is caused by the missing dependency of `openresty-openssl111-devel`, You need to install it first. Please refer to the [install-dependencies.md](doc/install-dependencies.md) document for installation.
```
Comment (tokers) [inline FAQ.md:361]: `You` should in small case, the sentence doesn't finish.

Comment (Firstsawyou) [reply FAQ.md]: Nice, updated.

File: FAQ.md [hunk 3]
```diff
@@ -355,3 +355,19 @@ The high availability of APISIX can be divided into two parts:
 1. The data plane of Apache APISIX is stateless and can be elastically scaled at will. Just add a layer of LB in front.
 
 2. The control plane of Apache APISIX relies on the highly available implementation of `etcd cluster` and does not require any relational database dependency.
+
+## Why does the `make deps` command fail in source installation?
+
+When executing the `make deps` command, an error such as the one shown below occurs. This is caused by the missing dependency of `openresty-openssl111-devel`, you need to install it first. Please refer to the [install-dependencies.md](doc/install-dependencies.md) document for installation.
```
Comment (tokers) [inline FAQ.md:361]: In Debian/Ubuntu Linux, the package name is not `openresty-openssl111-devel` but `openresty-openssl111-dev`. Here we don't need to show the real package name but tell people the development package of openssl for OpenResty is missing, guiding them to refer to the document.

Comment (Firstsawyou) [reply FAQ.md]: updated.

**Review by tokers — APPROVED** [review]

**Comment by Firstsawyou** [comment]
@spacewander  If you have time, please take a look.

**Review by spacewander — APPROVED** [review]

**Pull request merged**